### PR TITLE
2023-10-12

### PR DESCRIPTION
**Pull request by vmorsell**

### CODE REVIEW

File: 2023-10-12/main.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/vmorsell/joel/2023-10-12/names"
```
Comment (JoelGitKenobi) [inline 2023-10-12/main.go:6]: Importing the file names

File: 2023-10-12/main.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+)
+
+func main() {
+	fmt.Println(names.GitKenobi)
```
Comment (JoelGitKenobi) [inline 2023-10-12/main.go]: function from package fmt, Println, referring to file names, variable gitkenobi

File: 2023-10-12/names/names.go [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package names
```
Comment (JoelGitKenobi) [inline 2023-10-12/names/names.go:1]: defining package names which are referred to in the Println function

File: 2023-10-12/names/names.go [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
+
+var GitKenobi = "Joel"
```
Comment (JoelGitKenobi) [inline 2023-10-12/names/names.go:3]: defining variable GitKenobi setting it to string

**Review by JoelGitKenobi — APPROVED** [review]
Result: at run will provide value from from variable Gitkenobi, in file names. Variable is set as "Joel" therefore result prints: Joel

File: 2023-10-12/main.go (hunk 1 above)
Comment (vmorsell) [reply 2023-10-12/main.go:6]: Almost, it's importing the _package_ names. Go actually doesn't care about file names. See what happens if you would change the name of the file _names.go_, or the directory _names_
https://go.dev/doc/code#Organization

File: 2023-10-12/main.go (hunk 2 above)
Comment (vmorsell) [reply 2023-10-12/main.go]: Calling the exported function Println from the package _fmt_. First and only argument is the exported variable GitKenobi from the package _names_

File: 2023-10-12/main.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+	fmt.Scan(&name)
+
+	if name == names.GitKenobi || name == names.Vmorsell {
+		fmt.Printf("Welcome, %s \n",
+			name)
+	} else {
+		fmt.Println("You don't live here, leave")
+	}
+}
```
Comment (vmorsell) [inline 2023-10-12/main.go:23]: I reverted 2023-10-12 to the original and moved this to 2023-10-12_2 in 7aadb0ddd61a73e270f685ed1fed2bb0f8ce9d80 and a22c78fc706c850b31eee69d31e8d9b266bc24f6

**Pull request merged**